src/USER-ADIOS/dump_atom_adios.h

```diff
@@ -0,0 +1,70 @@
+/* -*- c++ -*- ----------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#ifdef DUMP_CLASS
+
+DumpStyle(atom/adios,DumpAtomADIOS)
+
+#else
+
+#ifndef LMP_DUMP_ATOM_ADIOS_H
+#define LMP_DUMP_ATOM_ADIOS_H
+
+#include "dump_atom.h"
+#include <stdlib.h>
+#include <stdint.h>
+#include "adios2.h"
+
+namespace LAMMPS_NS {
+
+class DumpAtomADIOS : public DumpAtom {
+
+ public:
+  DumpAtomADIOS(class LAMMPS *, int, char **);
+  virtual ~DumpAtomADIOS();
+
+ protected:
+
+  const std::string ioName="atom";   // name of adios group, referrable in adios2_config.xml
+  adios2::ADIOS *ad = nullptr; // adios object
```

Comment:
for "style" class headers, please hide all implementation specific data elements into a wrapper struct or class, and only store a pointer to that kind of struct/class, so that we don't need to include any of those headers here and can instead use a forward declaration and pass this as an opaque pointer, if needed. All "style" headers need to be included at the same time, and thus we need to reduce any potential name space collisions or inconsistencies due to the order in which non-LAMMPS headers are included (it has happened in the past repeatedly).